modules/snooker/src/test/ts/atomic/operate/TransformOperationsTest.ts

```diff
@@ -1,86 +1,117 @@
 import { assert, UnitTest } from '@ephox/bedrock-client';
-import { Arr, Fun } from '@ephox/katamari';
-import { SugarElement } from '@ephox/sugar';
+import { Arr } from '@ephox/katamari';
+import { SugarElement, SugarNode } from '@ephox/sugar';
 import { Generators } from 'ephox/snooker/api/Generators';
 import * as Structs from 'ephox/snooker/api/Structs';
 import * as TransformOperations from 'ephox/snooker/operate/TransformOperations';
+import * as MockStructs from 'ephox/snooker/test/MockStructs';
 import TestGenerator from 'ephox/snooker/test/TestGenerator';
 
 UnitTest.test('TransformOperationsTest', () => {
-  const en = (fakeElement: any, isNew: boolean) => Structs.elementnew(fakeElement as SugarElement, isNew);
+  let originalElements: Structs.ElementNew[] = [];
+  let expectedElements: Structs.ElementNew[] = [];
+
+  const clearElements = () => {
+    originalElements = [];
+    expectedElements = [];
+  };
+
+  // Original
+  const enO = (text: string, isNew: boolean, elemType: keyof HTMLElementTagNameMap = 'td') =>
+    MockStructs.getElementNew(originalElements, elemType, text, isNew);
+
+  // Expected
+  const enE = (text: string, isNew: boolean, elemType: keyof HTMLElementTagNameMap = 'td') =>
+    MockStructs.getElementNew(expectedElements, elemType, text, isNew);
 
   const mapToStructGrid = (grid: Structs.ElementNew[][]) => {
     return Arr.map(grid, (row) => {
-      return Structs.rowcells(row, 'tbody');
+      const hasCol = Arr.exists(row, (elementNew) => SugarNode.isTag('col')(elementNew.element));
+      return Structs.rowcells(row, hasCol ? 'colgroup' : 'tbody');
     });
   };
 
   const assertGrids = (expected: Structs.RowCells[], actual: Structs.RowCells[]) => {
     assert.eq(expected.length, actual.length);
     Arr.each(expected, (row, i) => {
       Arr.each(row.cells, (cell, j) => {
-        assert.eq(cell.element, actual[i].cells[j].element);
-        assert.eq(cell.isNew, actual[i].cells[j].isNew);
+        assert.eq(
+          cell.element.dom.innerText,
+          actual[i].cells[j].element.dom.innerText,
+          `innerText expected: "${cell.element.dom.innerText}". actual: "${actual[i].cells[j].element.dom.innerText}"`);
```

Comment:
```suggestion
        const actualCell = actual[i].cells[j];
        assert.eq(
          TextContent.get(actualCell.element),
          TextContent.get(cell.element),
          `innerText expected: "${TextContent.get(cell.element)}". actual: "${TextContent.get(actualCell.element)}"`);
```
This assertion was backwards, as `actual` needed to be first and `expected` second. We also should be using Sugar APIs where possible, so just swapped to using `TextContent`.

Reply:
The old assert has `expected` first, should I just use chai here?

Reply:
I think the old assert just has the incorrect names, as it was meant to be a replacement for similar assert libraries and they almost always have `actual` first. Looking at the code, it doesn't actually care which why they are used. In saying that though, I'm happy to switch to chai for this if others are.

Reply:
I vote switching to chai